src/pages/[id].js

```diff
@@ -1,54 +1,78 @@
 import { useRouter } from 'next/router';
 import { useEffect, useState } from 'react';
-import axios from 'axios';
 import { Typography, Card, CardContent, CardMedia, Box } from '@mui/material';
+import 'react-toastify/dist/ReactToastify.css';
+import { ToastContainer, toast } from 'react-toastify';
 
 const ProductDetails = () => {
   const [product, setProduct] = useState(null);
+  const [loading, setLoading] = useState(true);
   const router = useRouter();
   const { id } = router.query;
 
   useEffect(() => {
-    if (id) {
-      axios.get(`https://fakestoreapi.com/products/${id}`)
-        .then(response => {
-          setProduct(response.data);
-        })
-        .catch(error => {
-          console.error('Error fetching product details:', error);
-        });
-    }
+    const fetchData = async () => {
+      try {
+        if (id) {
+          const response = await fetch(`https://fakestoreapi.com/products/${id}`);
```

Comment:
store 
https://fakestoreapi.com/
this as base url 
make a Config.js file then import it here